src/Lykke.Service.IroncladDecorator/Sessions/UserSession/IdentityToken.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using System.IdentityModel.Tokens.Jwt;
+using System.Linq;
+using IdentityModel;
+using MessagePack;
+
+namespace Lykke.Service.IroncladDecorator.Sessions
+{
+    [MessagePackObject(true)]
+    public class IdentityToken
```

Comment:
We can remove `IdentityToken` and replace it with `ClaimsPrincipal`. Or create a separate class called `User` which will encapsulate `ClaimsPrincipal` interaction.
Claims principal is returned from IdToken validation. This could be done in the validation task.
We only use `UserId` in one place to pass userId to a method call:
https://github.com/LykkeCorp/Lykke.Service.IroncladDecorator/pull/17/files#diff-a16b1a7a71fe654676d5e1b66fab1b3aR82

Reply:
Discussed